DDG4/hepmc/HepMC3EventReader.cpp

```diff
@@ -130,12 +130,24 @@ HEPMC3EventReader::readParticles(int event_number, Vertices& vertices, Particles
     p->genStatus = genStatus&G4PARTICLE_GEN_STATUS_MASK;
 
     if ( p->parents.size() == 0 )  {
+      // This is a bit unintuitive. A particle without a parent in HepMC
+      // can only be (something like) a beam particle, and it is attached
+      // to the root vertex, by default (0,0,0) and equal for all parent-less particles.
+      // Therefore, p->vsx etc. is not suitable to cover situations with multiple
+      // particles coming from different vertices.
+      // Intuitively, one should add vertices for all particles that have a parent,
```

Comment:
If you want your pre-assigned decays to be consistent, you cannot add G4Vertices for each particle that has a parent.